src/synth/fluid_synth.c

```diff
@@ -408,6 +409,18 @@ fluid_synth_init(void)
 		       );
   fluid_mod_set_dest(&default_pitch_bend_mod, GEN_PITCH);                 /* Destination: Initial pitch */
   fluid_mod_set_amount(&default_pitch_bend_mod, 12700.0);                 /* Amount: 12700 cents */
+
+
+  /* Non-standard MIDI continuous controller 8 to channel stereo balance */
+  fluid_mod_set_source1(&default_balance_mod, BALANCE_MSB, /* Index=8 */
+		       FLUID_MOD_CC                              /* CC=1 */
+		       | FLUID_MOD_LINEAR                        /* type=0 */
+		       | FLUID_MOD_BIPOLAR                       /* P=1 */
+		       | FLUID_MOD_POSITIVE                      /* D=0 */
+		       );
+  fluid_mod_set_source2(&default_balance_mod, 0, 0);
+  fluid_mod_set_dest(&default_balance_mod, GEN_BALANCE);                  /* Destination: stereo balance */
+  fluid_mod_set_amount(&default_balance_mod, 1000.0);                     /* Amount: 1000 tens of a percent */
```

Comment:
Soundfont specifies that generator ranges should be in real-world units. -500;500 makes perfect sense for pan, because it is defined as the position on a 180 degree half circle. So -500 is 50% of 180 left.

Balance is not expressed as position, but as volume. So 100% left makes much more sense to say "right channel is silent" than 50%.

Reply:
Hm... Thinking about it, maybe balance volume should be expressed in centibels, like volume and expression. That would make the range -960;960, the modulator would get a concave shape and we would use fluid_cb2amp for conversion. Actually, that would also make volume changes much smoother in the extreme regions. What do you think?

Reply:
I just thought that giving it a pure linear scale wouldnt be "smooth" enough. I thought using a sinus would be still somewhat linear but in a smoother manner. This is where we would end up in fluid_pan() again.

fluid_cb2amp() could work as well. What I dont like about that [centi|deci]bel scale is that it never turns off. Yes, the volume gets very silent, probably unhearable, but if I set CC8=127 I would expect the left chan to be completely off.

But sure when looking at volume and expression, perhaps we should really use fluid_cb2amp() to be somewhat "consistent"?

Reply:
The sinus in pan is actually there to linearise the perceived volume, using equal power distribution. In listening tests I checked the perceived volume changes on a mono sample with pan and balance. With the current implementation, they are very similar, if not the same.

Using centibels, balance would be much smoother. Oh, and we could simply force -960 and 960 to be 0 (or 1 on the opposite channel)

Reply:
Alright then fluid_cb2amp().

> Oh, and we could simply force -960 and 960 to be 0 (or 1 on the opposite channel)

No special casing pls. 10^(960/-200)=0.000016 and that's how things should be then :)

Reply:
> No special casing pls. 10^(960/-200)=0.000016 and that's how things should be then :)

In that case I would strongly prefer the current solution. A balance control that can't turn be channel off completely would be flawed, imho.

What's so bad about forcing the extreme values to 0?

Reply:
> A balance control that can't turn be channel off completely would be flawed, imho.

Same like a main volume control that cant turn off a channel completely, which currently seems to be the case (at least I experienced that).

> What's so bad about forcing the extreme values to 0?

I dont like branches in math calculations. They cause hearable discontinuity and branch prediction failures. They just dont feel right. Where would you place these if clauses? Probably after modulator processing, a place where you would need to do a floating point comparison for `x==960.0`. How likely is that to evaluate to true?

> In that case I would strongly prefer the current solution.

Or we go with fluid_pan(), i.e. sin().